core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -755,6 +763,48 @@ object Chunk {
   def fromArray[A](array: Array[A]): Chunk[A] =
```

Comment:
Let's make this method do the right thing by pattern matching on the array. You can see an example for this in `WrappedArray.make`.